tetris.js

```diff
@@ -229,49 +242,63 @@ module.exports = {
 		return newArr;
 	},
 	
-	buildTetromino: function(stringOfCoordinates){
-		let math = stringOfCoordinates.split("\n")
-		let arrayOfCoordinates = []
+	getCoordinates: function(stringOfCoordinates){
+		let fieldArray
 		let quantityElem = 0;
+		let arrayOfCoordinates = []
+		if(typeof(stringOfCoordinates) == "string"){
+			fieldArray = stringOfCoordinates.split("\n")
+		}else{
+			fieldArray = stringOfCoordinates
+		}
 		
-		for(let i = 0, elem_Y = 0; i < math.length; i++){
-			if(math[i]!= false){
+		for(let i = 0, axisElement_Y = 0; i < fieldArray.length; i++){
+			if(fieldArray[i]!= false){
 				
-				for(let j = 0, elem_X = 0; j < math[i].length; j++){
+				for(let j = 0, axisElement_X = 0; j < fieldArray[i].length; j++){
 					let obj = {}
-					if(math[i][j]!= false){
-						if(math[i][j] == "X"){
-							console.log(math[i][j])
-							obj.x = elem_X
-							obj.y = elem_Y
+					if(fieldArray[i][j]!= false){
+						if(fieldArray[i][j] == "X"){
+							obj.x = axisElement_X
+							obj.y = axisElement_Y
 							quantityElem++;
 							
-							arrayOfCoordinates.push(obj)
+							arrayOfCoordinates.push(obj);
 							
 						}
-						elem_X++
+						axisElement_X++;
 					}
 					
 				}
-				elem_Y++
+				axisElement_Y++;
 			}
 			
 		}
+		return arrayOfCoordinates;
+	},
+	
+	buildTetromino: function(stringOfCoordinates){
+		let fieldArray = stringOfCoordinates.split("\n");
+		let arrayOfCoordinates = [];
+		module.exports.getCoordinates(fieldArray);
```

Comment:
1. where do you use the result of this function invocation? exact line, please
2. Are you changing fieldArray inside getCoordinates?

Reply:
line 279, 284, 285...

Reply:
1. Wrong
2. Please answer the second question

Reply:
1. Also 279 is not even line of code

Reply:
1. Hint: are you using the result of the invocation in line 283?

Reply:
I did